CustomLookup/CustomEntitySearch.cs

```diff
@@ -0,0 +1,187 @@
+            };
+
+            string requestBody = new StreamReader(req.Body).ReadToEnd();
+            var data = JsonConvert.DeserializeObject<WebApiRequest>(requestBody);
+                              
+            // Do some schema validation
+            if (data == null)
+            {
+                return new BadRequestObjectResult("The request schema does not match expected schema.");
+            }
+            if (data.Values == null)
+            {
+                return new BadRequestObjectResult("The request schema does not match expected schema. Could not find values array.");
+            }
+
+            // Calculate the response for each value.
+            foreach (var record in data.Values)
+            {
+                if (record == null || record.RecordId == null) continue;
+
+                if (record.Data.Text == null)
+                {
+                    return new BadRequestObjectResult("The request schema does not match expected schema. Could not find text string.");
+                }
+
+                if (record.Data.Words == null)
+                {
+                    return new BadRequestObjectResult("The request schema does not match expected schema. Could not find words array.");
+                }
+
+                OutputRecord responseRecord = new OutputRecord
+                {
+                    RecordId = record.RecordId
+                };
+
+                try
+                {
+                    responseRecord.Data = await FindEntity(record.Data.Text, record.Data.Words);
+                }
+                catch (Exception e)
+                {
+                    // Something bad happened, log the issue.
+                    return new BadRequestObjectResult(e.Message);
+                }
+                finally
+                {
+                    response.Values.Add(responseRecord);
+                }
+            }
+
+            return (ActionResult)new OkObjectResult(response);
+        }
+        #endregion
+
+        #region Helper functions (My main change to the sample code provided)
+        /// <summary>
+        /// Given text document and user-defined entities, this function determines if the provided
+        /// entities exist in the given text. If so, the first match's index is returned. Otherwise,
+        /// an index of "-1" is returned. Assumes user wants just the entity or entity followed by any word character.
+        /// </summary>
+        /// <param name="text"> Text document provided by user</param>
+        /// <param name="words"> User-defined entities </param>
+        /// <returns></returns>
+        private async static Task<List<OutputRecord.OutputRecordData>> FindEntity(string text, List<string> words)
+        {
+            List <OutputRecord.OutputRecordData>  data = new List<OutputRecord.OutputRecordData>();
+
+            string[] multiComp = new string[words.Count];
+            for (int i = 0; i < words.Count; i++)
```

Comment:
You could simplify that with `string[] escapedWords = words.Select(Regex.Escape).ToArray()`. (I also changed the name of that variable to something a little more expressive.

Reply:
Actually, there's no reason to allocate and keep this array around at all. You can escape each word as part of your second loop.